lib/generic-pool.js

```diff
@@ -2,78 +2,76 @@
  * @class
  * @private
  */
-function PriorityQueue(size) {
-
+function PriorityQueue (size) {
   if (!(this instanceof PriorityQueue)) {
-    return new PriorityQueue();
+    return new PriorityQueue()
   }
 
-  this._size = size;
-  this._slots = null;
-  this._total = null;
+  this._size = size
+  this._slots = null
+  this._total = null
 
   // initialize arrays to hold queue elements
-  size = Math.max(+size | 0, 1);
-  this._slots = [];
+  size = Math.max(+size | 0, 1)
+  this._slots = []
   for (var i = 0; i < size; i += 1) {
-    this._slots.push([]);
+    this._slots.push([])
   }
-};
+}
 
-PriorityQueue.prototype.size = function size() {
+PriorityQueue.prototype.size = function size () {
   if (this._total === null) {
-    this._total = 0;
+    this._total = 0
     for (var i = 0; i < this._size; i += 1) {
-      this._total += this._slots[i].length;
+      this._total += this._slots[i].length
     }
   }
-  return this._total;
-};
+  return this._total
+}
 
-PriorityQueue.prototype.enqueue = function enqueue(obj, priority) {
-  var priorityOrig;
+PriorityQueue.prototype.enqueue = function enqueue (obj, priority) {
+  var priorityOrig
 
   // Convert to integer with a default value of 0.
-  priority = priority && + priority | 0 || 0;
+  priority = priority && + priority | 0 || 0 // eslint-disable-line space-unary-ops
```

Comment:
Because I'm not entirely sure quite how this resolves and operator precedence and didn't know how to make it less opaque at the time, it seems like an overly clever way of achieving it's a goal. So I just bailed and disabled the checking :-p 

Any input / improvements / explanation gladly excepted!